### DISCUSSION TITLE
Semantics of the messages in `expect`

### DISCUSSION THREAD

**Discussion by elBoberido** [eclipse-iceoryx/iceoryx#2029] [General]
In my last PR is described the actual expected result in the `expect` call of the `expected` and `optional`, e.g. `This is okay since Foo is guaranteed` but I was made aware that in iceoryx it was mainly used to describe the unexpected outcome, e.g. `Failed because Foo invariant is broken`. From my experience with Rust I'm more used to the former semantics, i.e. describe what is expected to happen. This can also be used instead of comments and as justification why `expect` is called instead of doing some error handling.

What do you think about changing the semantics of the `expect` message to describe why it is safe to call `expect` and also adjust the log output to something like
```
[FATAL] Expected 'This is okay since Foo is guaranteed' but failed
```

I would volunteer to change the current strings in iceoryx.

cc @elfenpiff @FerdinandSpitzschnueffler @mossmaurice @dkroenke @MatthiasKillat

**Comment by elfenpiff** [comment]
@elBoberido I would also prefer the rust semantics and would support this.

**Reply by elBoberido** [reply]
@elfenpiff okay, then we have already 4 committer supporting this. Will create an issue and open an PR on a rainy day

**Comment by elBoberido** [comment]
Created #2036
